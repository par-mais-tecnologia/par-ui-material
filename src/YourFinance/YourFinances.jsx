import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Grid, Typography } from '../index'
import { getCurrencyFormat } from '../Core/masks'

class YourFinances extends PureComponent {
  render () {
    const { incomes, expenses } = this.props
    const balance = incomes - expenses

    let percentBalance = 0
    let percentIncomes = 100
    let percentExpenses = 100

    if (incomes > expenses) {
      percentIncomes = 100
      percentExpenses = Number(((expenses / incomes) * 100).toFixed(1))
      percentBalance = Number((((incomes - expenses) / incomes) * 100).toFixed(1))
    } else if (expenses > incomes) {
      percentExpenses = 100
      percentIncomes = Number(((incomes / expenses) * 100).toFixed(1))
      percentBalance = Number((((expenses - incomes) / expenses) * 100).toFixed(1))
    }

    const barStyles = {
      barIncomes: {
        height: 2,
        backgroundColor: '#94BA1D',
        width: `${percentIncomes}%`
      },
      barExpenses: {
        height: 2,
        backgroundColor: '#BA2B31',
        width: `${percentExpenses}%`
      },
      barBalance: {
        height: 2,
        backgroundColor: '#007CA3',
        width: `${percentBalance}%`
      }
    }

    return (
      <Grid container direction='column' className='mb3'>
        <Grid container justify='space-between'>
          <Typography>Rendas </Typography>
          <Typography>{getCurrencyFormat(incomes, 'R$', 0, ' ')}</Typography>
        </Grid>
        <Grid style={barStyles.barIncomes} className='mb3 mt1' />

        <Grid container justify='space-between'>
          <Typography>Despesas</Typography>
          <Typography>{getCurrencyFormat(expenses, 'R$', 0, ' ')}</Typography>
        </Grid>
        <Grid style={barStyles.barExpenses} className='mb3 mt1' />

        { balance > 0
          ? <Grid>
            <Grid container justify='space-between'>
              <Typography>
                <b>Saldo para investir </b>
              </Typography>
              <Typography>
                <b>{getCurrencyFormat(incomes - expenses, 'R$', 0, ' ')}</b>
              </Typography>
            </Grid>
            <Grid style={barStyles.barBalance} className='mt1' />
          </Grid>
          : <Grid className='mt2'>
            <Typography variant='body1' className='without-balance'>
              <b>Você não possui saldo para investir</b>
            </Typography>
          </Grid>}

      </Grid>
    )
  }
}

YourFinances.propTypes = {
  incomes: PropTypes.number,
  expenses: PropTypes.number
}

YourFinances.defaultProps = {
  incomes: 0,
  expenses: 0
}

export default YourFinances
