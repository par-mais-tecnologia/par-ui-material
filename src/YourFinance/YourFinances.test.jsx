import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import YourFinances from './'

configure({ adapter: new Adapter() })

describe('YourFinances', () => {
  it('shallow YourFinances', () => {
    const wrapper = shallow(
      <YourFinances
        incomes={5000}
        expenses={2000}
      />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow YourFinances match Snapshot', () => {
    const wrapper = shallow(
      <YourFinances
        incomes={4500}
        expenses={1000}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should inform that there is not sufficient balance for investments', () => {
    const wrapper = shallow(
      <YourFinances
        incomes={2500}
        expenses={2600}
      />
    )
    expect(wrapper.find('.without-balance')).toHaveLength(1)
  })
})
