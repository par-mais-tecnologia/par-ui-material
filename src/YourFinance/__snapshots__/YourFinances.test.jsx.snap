// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourFinances shallow YourFinances match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YourFinances
    expenses={1000}
    incomes={4500}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Grid
          container={true}
          justify="space-between"
        >
          <Typography>
            Rendas 
          </Typography>
          <Typography>
            R$ 4,500 00
          </Typography>
        </Grid>,
        <Grid
          className="mb3 mt1"
          style={
            Object {
              "backgroundColor": "#94BA1D",
              "height": 2,
              "width": "100%",
            }
          }
        />,
        <Grid
          container={true}
          justify="space-between"
        >
          <Typography>
            Despesas
          </Typography>
          <Typography>
            R$ 1,000 00
          </Typography>
        </Grid>,
        <Grid
          className="mb3 mt1"
          style={
            Object {
              "backgroundColor": "#BA2B31",
              "height": 2,
              "width": "22.2%",
            }
          }
        />,
        <Grid>
          <Grid
            container={true}
            justify="space-between"
          >
            <Typography>
              <b>
                Saldo para investir 
              </b>
            </Typography>
            <Typography>
              <b>
                R$ 3,500 00
              </b>
            </Typography>
          </Grid>
          <Grid
            className="mt1"
            style={
              Object {
                "backgroundColor": "#007CA3",
                "height": 2,
                "width": "77.8%",
              }
            }
          />
        </Grid>,
      ],
      "className": "mb3",
      "container": true,
      "direction": "column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Typography>
              Rendas 
            </Typography>,
            <Typography>
              R$ 4,500 00
            </Typography>,
          ],
          "container": true,
          "justify": "space-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Rendas ",
            },
            "ref": null,
            "rendered": "Rendas ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "R$ 4,500 00",
            },
            "ref": null,
            "rendered": "R$ 4,500 00",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "mb3 mt1",
          "style": Object {
            "backgroundColor": "#94BA1D",
            "height": 2,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Typography>
              Despesas
            </Typography>,
            <Typography>
              R$ 1,000 00
            </Typography>,
          ],
          "container": true,
          "justify": "space-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Despesas",
            },
            "ref": null,
            "rendered": "Despesas",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "R$ 1,000 00",
            },
            "ref": null,
            "rendered": "R$ 1,000 00",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "mb3 mt1",
          "style": Object {
            "backgroundColor": "#BA2B31",
            "height": 2,
            "width": "22.2%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Grid
              container={true}
              justify="space-between"
            >
              <Typography>
                <b>
                  Saldo para investir 
                </b>
              </Typography>
              <Typography>
                <b>
                  R$ 3,500 00
                </b>
              </Typography>
            </Grid>,
            <Grid
              className="mt1"
              style={
                Object {
                  "backgroundColor": "#007CA3",
                  "height": 2,
                  "width": "77.8%",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Typography>
                  <b>
                    Saldo para investir 
                  </b>
                </Typography>,
                <Typography>
                  <b>
                    R$ 3,500 00
                  </b>
                </Typography>,
              ],
              "container": true,
              "justify": "space-between",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <b>
                    Saldo para investir 
                  </b>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Saldo para investir ",
                  },
                  "ref": null,
                  "rendered": "Saldo para investir ",
                  "type": "b",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <b>
                    R$ 3,500 00
                  </b>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "R$ 3,500 00",
                  },
                  "ref": null,
                  "rendered": "R$ 3,500 00",
                  "type": "b",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "mt1",
              "style": Object {
                "backgroundColor": "#007CA3",
                "height": 2,
                "width": "77.8%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Grid
            container={true}
            justify="space-between"
          >
            <Typography>
              Rendas 
            </Typography>
            <Typography>
              R$ 4,500 00
            </Typography>
          </Grid>,
          <Grid
            className="mb3 mt1"
            style={
              Object {
                "backgroundColor": "#94BA1D",
                "height": 2,
                "width": "100%",
              }
            }
          />,
          <Grid
            container={true}
            justify="space-between"
          >
            <Typography>
              Despesas
            </Typography>
            <Typography>
              R$ 1,000 00
            </Typography>
          </Grid>,
          <Grid
            className="mb3 mt1"
            style={
              Object {
                "backgroundColor": "#BA2B31",
                "height": 2,
                "width": "22.2%",
              }
            }
          />,
          <Grid>
            <Grid
              container={true}
              justify="space-between"
            >
              <Typography>
                <b>
                  Saldo para investir 
                </b>
              </Typography>
              <Typography>
                <b>
                  R$ 3,500 00
                </b>
              </Typography>
            </Grid>
            <Grid
              className="mt1"
              style={
                Object {
                  "backgroundColor": "#007CA3",
                  "height": 2,
                  "width": "77.8%",
                }
              }
            />
          </Grid>,
        ],
        "className": "mb3",
        "container": true,
        "direction": "column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Typography>
                Rendas 
              </Typography>,
              <Typography>
                R$ 4,500 00
              </Typography>,
            ],
            "container": true,
            "justify": "space-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Rendas ",
              },
              "ref": null,
              "rendered": "Rendas ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "R$ 4,500 00",
              },
              "ref": null,
              "rendered": "R$ 4,500 00",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "mb3 mt1",
            "style": Object {
              "backgroundColor": "#94BA1D",
              "height": 2,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Typography>
                Despesas
              </Typography>,
              <Typography>
                R$ 1,000 00
              </Typography>,
            ],
            "container": true,
            "justify": "space-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Despesas",
              },
              "ref": null,
              "rendered": "Despesas",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "R$ 1,000 00",
              },
              "ref": null,
              "rendered": "R$ 1,000 00",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "mb3 mt1",
            "style": Object {
              "backgroundColor": "#BA2B31",
              "height": 2,
              "width": "22.2%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Grid
                container={true}
                justify="space-between"
              >
                <Typography>
                  <b>
                    Saldo para investir 
                  </b>
                </Typography>
                <Typography>
                  <b>
                    R$ 3,500 00
                  </b>
                </Typography>
              </Grid>,
              <Grid
                className="mt1"
                style={
                  Object {
                    "backgroundColor": "#007CA3",
                    "height": 2,
                    "width": "77.8%",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Typography>
                    <b>
                      Saldo para investir 
                    </b>
                  </Typography>,
                  <Typography>
                    <b>
                      R$ 3,500 00
                    </b>
                  </Typography>,
                ],
                "container": true,
                "justify": "space-between",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <b>
                      Saldo para investir 
                    </b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Saldo para investir ",
                    },
                    "ref": null,
                    "rendered": "Saldo para investir ",
                    "type": "b",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <b>
                      R$ 3,500 00
                    </b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "R$ 3,500 00",
                    },
                    "ref": null,
                    "rendered": "R$ 3,500 00",
                    "type": "b",
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "mt1",
                "style": Object {
                  "backgroundColor": "#007CA3",
                  "height": 2,
                  "width": "77.8%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
